site/reference/fn/eval/README.md

```diff
@@ -0,0 +1,133 @@
+---
+title: "`eval`"
+linkTitle: "eval"
+type: docs
+description: >
+  Execute function on resources
+---
+
+<!--mdtogo:Short
+    Execute function on resources
+-->
+
+`eval` executes function on resources in a directory and writes the output to
+the local filesystem in-place. Functions are packaged as container images or
+binary executables.
+
+If the function fails (i.e. exits with non-zero status code), `eval` will
+abort and the local filesystem is left intact.
+
+Refer to the [Imperative Function Execution] for detailed overview.
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+```shell
+kpt fn eval [DIR|-] [flags]
+```
+
+#### Args
+
+```shell
+DIR|-:
+  Path to the local directory containing resources. Defaults to the current
+  working directory. Using '-' as the directory path will cause `eval` to
+  read resources from `stdin` and write the output to `stdout`.
+```
+
+#### Flags
+
+```shell
+--image:
+  Container image of the function to execute e.g. `gcr.io/kpt-fn/set-namespace:v0.1`
+
+--exec-path:
+  Path to the local executable binary to execute as a function.
+  
+--fn-config:
+  Path to the file containing `functionConfig` for the function.
+
+--include-meta-resources:
+  If enabled, meta resources (i.e. `Kptfile` and `functionConfig`) are included
+  in the input to the function. By default it is disabled.
+
+--network:
+  If enabled, container functions are allowed to access network.
+  By default is it disabled.
+
+--mount:
+  List of storage options to enable reading from the local filesytem. By default,
+  container functions can not access the local filesystem. It accepts the same options
+  as specified on the [Docker Volumes] for `docker run`. All volumes are mounted
+  readonly by default. Specify `rw=true` to mount volumes in read-write mode.
+
+--env:
+  List of local environment variables to be exported to the container function.
+  By default, none of local environment variables are made available to the
+  container running the function. The value can be in `key=value` format or only
+  the key of an already exported environment variable.
+
+--as-current-user:
+  Use the `uid` and `gid` of the kpt process for container function execution.
```

Comment:
We should document the where to use this flag. Otherwise user will find the mounted directory is not readable from the container on Linux.

Reply:
Added a note. Also added an example.